widgets/nuxeo-uploader-behavior.html

```diff
@@ -113,113 +274,87 @@
             console.log('Batch Execute operation failed: ' + error);
             throw error;
           }.bind(this));
-      }.bind(this));
-    },
+      },
 
-    cancelBatch: function() {
-      if (this.uploader) {
-        if (this.uploader._batchId) {
-          this.uploader.cancel();
+      cancelBatch: function() {
+        return this._providerInstance.cancelBatch();
+      },
+
+      hasProgress: function() {
+        return this._providerInstance.hasProgress();
+      },
+
+      accepts: function(files) {
+        if (!this._providerInstance) {
+          return false;
         }
-        this.uploader = null;
-        this.batchId = null;
-      }
-    },
+        if (files.length) {
```

Comment:
`if (files)`?

Reply:
Ok, I see you are checking for `length` to see if it's an array or not? if that's the case, imo it's a very misleading API. I would assume `files` is always an `Array`.

Reply:
Its because the method can take either an array of files or a file. Probably legacy behavior, but it's a public method and perhaps we should keep the API intact.